sdk/eventhub/azure-eventhubs/azure/eventhub/_client_base.py

```diff
@@ -0,0 +1,376 @@
+        raise last_exception
+
+    def _add_span_request_attributes(self, span):
+        span.add_attribute("component", "eventhubs")
+        span.add_attribute("message_bus.destination", self._address.path)
+        span.add_attribute("peer.address", self._address.hostname)
+
+    def get_properties(self):
+        # type:() -> Dict[str, Any]
+        """
+        Get properties of the specified EventHub async.
+        Keys in the details dictionary include:
+
+            - path
+            - created_at
+            - partition_ids
+
+        :rtype: dict
+        :raises: :class:`EventHubError<azure.eventhub.EventHubError>`
+        """
+        mgmt_msg = Message(application_properties={'name': self.eh_name})
+        response = self._management_request(mgmt_msg, op_type=MGMT_OPERATION)
+        output = {}
+        eh_info = response.get_data()
+        if eh_info:
+            output['path'] = eh_info[b'name'].decode('utf-8')
+            output['created_at'] = datetime.datetime.utcfromtimestamp(float(eh_info[b'created_at']) / 1000)
+            output['partition_ids'] = [p.decode('utf-8') for p in eh_info[b'partition_ids']]
+        return output
+
+    def get_partition_ids(self):
+        # type:() -> List[str]
+        """
+        Get partition ids of the specified EventHub.
+
+        :rtype: list[str]
+        :raises: :class:`EventHubError<azure.eventhub.EventHubError>`
+        """
+        return self.get_properties()['partition_ids']
+
+    def get_partition_properties(self, partition):
+        # type:(str) -> Dict[str, Any]
+        """
+        Get properties of the specified partition async.
+        Keys in the details dictionary include:
+
+            - event_hub_path
+            - id
+            - beginning_sequence_number
+            - last_enqueued_sequence_number
+            - last_enqueued_offset
+            - last_enqueued_time_utc
+            - is_empty
+
+        :param partition: The target partition id.
+        :type partition: str
+        :rtype: dict
+        :raises: :class:`EventHubError<azure.eventhub.EventHubError>`
+        """
+        mgmt_msg = Message(application_properties={'name': self.eh_name,
+                                                   'partition': partition})
+        response = self._management_request(mgmt_msg, op_type=MGMT_PARTITION_OPERATION)
+        partition_info = response.get_data()
+        output = {}
+        if partition_info:
+            output['event_hub_path'] = partition_info[b'name'].decode('utf-8')
+            output['id'] = partition_info[b'partition'].decode('utf-8')
+            output['beginning_sequence_number'] = partition_info[b'begin_sequence_number']
+            output['last_enqueued_sequence_number'] = partition_info[b'last_enqueued_sequence_number']
+            output['last_enqueued_offset'] = partition_info[b'last_enqueued_offset'].decode('utf-8')
+            output['last_enqueued_time_utc'] = datetime.datetime.utcfromtimestamp(
```

Comment:
use `datetime.fromtimestamp` with `tz`. See the [warning in the documentation](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp).